homeassistant/components/iqvia/__init__.py

```diff
@@ -103,94 +106,75 @@ def __init__(self, client, sensor_types):
         self.sensor_types = sensor_types
         self.zip_code = client.zip_code
 
-    async def _get_data(self, method, key):
-        """Return API data from a specific call."""
-        from pyiqvia.errors import IQVIAError
-
-        try:
-            data = await method()
-            self.data[key] = data
-        except IQVIAError as err:
-            _LOGGER.error('Unable to get "%s" data: %s', key, err)
-            self.data[key] = {}
+        FETCHERS.register(TYPE_ALLERGY_FORECAST)(
```

Comment:
Since we update the registry here, it shouldn't be a constant. Make it an instance attribute instead.